src/tests/helpers.ts

```diff
@@ -103,21 +104,23 @@ export function mockAPICalls(testName: string, url = MainnetUrl) {
 
 /**
  * @internal
- * returns either a standard mainnet infura url
+ * returns either a standard mainnet linkpool url
+ * @see https://docs.linkpool.io/docs/rpc_main
  * or the one with attached INFURA SECRET key from
  * UNSTOPPABLE_RESOLUTION_INFURA_PROJECTID env variable if any
  */
-export function secretInfuraLink(infuraProtocol: InfuraProtocol = InfuraProtocol.http): string {
+export function secretInfuraLink(providerProtocol: ProviderProtocol = ProviderProtocol.http): string {
   const secret = process.env.UNSTOPPABLE_RESOLUTION_INFURA_PROJECTID;
   const protocolMap = {
-    [InfuraProtocol.http]:'https://mainnet.infura.io/v3',
-    [InfuraProtocol.wss]:'wss://mainnet.infura.io/ws/v3'
+    [ProviderProtocol.http]:'https://mainnet.infura.io/v3',
+    [ProviderProtocol.wss]:'wss://mainnet.infura.io/ws/v3'
   };
-  const url = `${protocolMap[infuraProtocol]}/${secret}`;
+  let url = secret ? `${protocolMap[providerProtocol]}/${secret}` : 'https://main-rpc.linkpool.io';
+  if (!secret && providerProtocol === ProviderProtocol.wss) url = "wss://main-rpc.linkpool.io/ws";
```

Comment:
Two lines above make me think you are doing somethign really complex in this function. You don't in fact.

All you need is handle a matrix of 2x2 use cases: when `secret` is defined and not defined and when protocol is `http` or `wss`. This should be as easy as:

```  typescript
return {  
    [ProviderProtocol.http]: secret ? `https://mainnet.infura.io/v3/${secret}`  : 'https://main-rpc.linkpool.io',
    [InfuraProtocol.wss]:    secret ? `wss://mainnet.infura.io/ws/v3/${secret}` : "wss://main-rpc.linkpool.io/ws"
}[providerProtocol]